components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice_test.go

```diff
@@ -0,0 +1,608 @@
+				kernelFilepath: failureKernelfile,
+			},
+			expectedBody: &models.SoftwareVersionDetails{
+				Passed: false,
+				Checks: []models.Checks{
+					{
+						Title:         StatTitle,
+						Passed:        true,
+						SuccessMsg:    StatIsAvailable,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         MkdirTitle,
+						Passed:        true,
+						SuccessMsg:    MkdirIsAvailable,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         KernalVersionTitle,
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      "Linux kernel version is lower than 3.2",
+						ResolutionMsg: "Use a linux version whose kernel version is greater than 3.2",
+					},
+					{
+						Title:         LinuxVersionTitle,
+						Passed:        true,
+						SuccessMsg:    "Ubuntu version is 20.04",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			expectedError: "",
+		},
+		{
+			description: "If file is not present for reading the kernal version",
+			args: args{
+				query: "postgres",
+				checkarray: checktrue,
+				osFilepath: successfile,
+				kernelFilepath: failfilepath,
+			},
+			expectedBody: &models.SoftwareVersionDetails{
+				Passed: false,
+				Checks: []models.Checks{
+					{
+						Title:         StatTitle,
+						Passed:        true,
+						SuccessMsg:    StatIsAvailable,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         MkdirTitle,
+						Passed:        true,
+						SuccessMsg:    MkdirIsAvailable,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         KernalVersionTitle,
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      "Its not feasible to determine the Kernal version of the system",
+						ResolutionMsg: "Please run automate on the supported platforms.",
+					},
+					{
+						Title:         LinuxVersionTitle,
+						Passed:        true,
+						SuccessMsg:    "Ubuntu version is 20.04",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			expectedError: "Error while getting the Kernal Version: ./failfilepath doesnot exist",
+		},
+		{
+			description: "If the entered Query is not supported by us",
+			args: args{
+				query:      "wrong-query",
+				checkarray: checktrue,
+				osFilepath: successfile,
+				kernelFilepath: successKernelfile,
+			},
+			expectedBody:  nil,
+			expectedError: "The query wrong-query is not supported. The Supported query's are: postgres, opensearch, bastion, automate, chef-server",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			service.(*SoftwareVersionService).osFilePath = tt.args.osFilepath
+			service.(*SoftwareVersionService).kernelFilePath = tt.args.kernelFilepath
+			service.(*SoftwareVersionService).cmdCheckArray = tt.args.checkarray
+			got, err := service.GetSoftwareVersionDetails(tt.args.query)
```

Comment:
same here, you need to assert based on the input

Reply:
Here removed the error column as we discussed my function gives the nil handled all the error cases in there personal defination